packages/cache/src/Cache.php

```diff
@@ -6,20 +6,88 @@
 
 use Closure;
 use Psr\Cache\CacheItemInterface;
+use Stringable;
 use Tempest\DateTime\DateTimeInterface;
+use Tempest\DateTime\Duration;
 
 interface Cache
 {
-    public function put(string $key, mixed $value, ?DateTimeInterface $expiresAt = null): CacheItemInterface;
+    /**
+     * Whether the cache is enabled.
+     */
+    public bool $enabled {
+        get;
+        set;
+    }
 
-    public function get(string $key): mixed;
+    /**
+     * Returns a lock for the specified key. The lock is not acquired until `acquire()` is called.
+     *
+     * @param Stringable|string $key The identifier of the lock.
+     * @param null|Duration|DateTimeInterface $expiration The expiration time for the lock. If not specified, the lock will not expire.
+     * @param null|Stringable|string $owner The owner of the lock, which will be used to identify the process releasing it. If not specified, a random string will be used.
+     */
+    public function lock(Stringable|string $key, null|Duration|DateTimeInterface $expiration = null, null|Stringable|string $owner = null): Lock;
 
-    /** @param Closure(): mixed $cache */
-    public function resolve(string $key, Closure $cache, ?DateTimeInterface $expiresAt = null): mixed;
+    /**
+     * Sets the specified key to the specified value in the cache. Optionally, specify an expiration.
+     */
+    public function put(Stringable|string $key, mixed $value, null|Duration|DateTimeInterface $expiration = null): CacheItemInterface;
 
-    public function remove(string $key): void;
+    /**
+     * Sets the specified keys to the specified values in the cache. Optionally, specify an expiration.
+     *
+     * @template TKey of array-key
+     * @template TValue
+     *
+     * @param iterable<TKey,TValue> $array
+     * @return array<TKey,CacheItemInterface>
+     */
+    public function putMany(iterable $values, null|Duration|DateTimeInterface $expiration = null): array;
 
-    public function clear(): void;
+    /**
+     * Gets the value associated with the specified key from the cache. If the key does not exist, null is returned.
+     */
+    public function get(Stringable|string $key): mixed;
+
+    /**
+     * Gets the values associated with the specified keys from the cache. If a key does not exist, null is returned for that key.
+     *
+     * @template TKey of array-key
+     * @template TValue
+     *
+     * @param iterable<TKey,TValue> $array
+     * @return array<TValue,mixed>
+     */
+    public function getMany(iterable $key): array;
+
+    /**
+     * Determines whether the cache contains the specified key.
+     */
+    public function has(Stringable|string $key): bool;
+
+    /**
+     * Increments the value associated with the specified key by the specified amount. If the key does not exist, it is created with the specified amount.
+     */
+    public function increment(Stringable|string $key, int $by = 1): int;
```

Comment:
Same for these methods. I'm not sure if they are really needed…

But please tell me I'm wrong :)

Reply:
Those however, I have used them! It's so much nicer than having to fetch the value, increment it yourself and update it back.

Reply:
Yeah I agree that I got pretty excited reading about these :p Let's keep them in